lib/surface/compiler.ex

```diff
@@ -1192,7 +1192,11 @@ defmodule Surface.Compiler do
     end
   end
 
-  defp validate_slot_attrs!([{name, _, %{file: file, line: line}} | _]) when name != "name" do
+  defp validate_slot_attrs!(attrs) do
+    Enum.each(attrs, &validate_slot_attr!/1)
+  end
+
+  defp validate_slot_attr!({name, _, %{file: file, line: line}}) when name not in ["name", "index"] do
```

Comment:
I think it would read more naturally if we avoid the negation and switch it around:
- when name in ["name", "index"] -> :ok
- rest -> raise compile error